chaco/scales/tests/test_time_scale.py

```diff
@@ -1,61 +1,240 @@
 
 
-import time
+import datetime
+import os
+import contextlib
 
 import numpy as np
 
-from chaco.scales.time_scale import tfrac, trange, \
-        TimeScale, CalendarScaleSystem
+from chaco.scales.time_scale import (
+    tfrac, trange, TimeScale, CalendarScaleSystem)
 from chaco.scales.api import TimeFormatter
 
+# Note on testing:
+# Chaco assumes times are in UTC seconds since Posix Epoch but does ticking
+# in whatever the local time is on the host machine.  This is problematic for
+# testing, because correct responses depend on current state of the test
+# machine, and will vary by location and time of year.  For hour/day/year
+# testing, where TZ matter, we select 3 timezones to test in:
+#
+# * UTC, because it's easy to reason about what the answer should be
+# * Australia/North (ie. Alice Springs, Darwin) which is UTC+9:30 and with no
+#   daylight savings time shifts, which should hopefully reveal bugs with
+#   non-integer hour offsets.
+# * Pacific/Honolulu which is UTC-10:00 and with no daylight savings time
+#   shifts, which should hopefully reveal bugs with previous-day or -year
+#   baselines.
+#
+# Note that this isn't quite bullet-proof, as things will break if Hawaii or
```

Comment:
From what I've read, using the `TZ` environment variable is unreliable in any context. I tested this on Windows, and it doesn't work.

AFAICT, this only seems to affect the `fromtimestamp(0.0)` below. If I'm right, then we can hardcode the `EPOCH` for each timezone that we care about and not worry about the environment variable.